tests/CMakeLists.txt

```diff
@@ -12,21 +12,25 @@ include_directories( SYSTEM $<TARGET_PROPERTY:dfelibs::dfelibs,INTERFACE_INCLUDE
 # Set up a test library, which the "new style" benchmarks and tests could use.
 file( GLOB _detray_test_headers
    RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
-   "include/detray/test/*.hpp"
-   "include/detray/test/detail/*.hpp" 
-   "include/detray/test/intersection/*.hpp"
-   "include/detray/test/utils/*.hpp")
+   "include/detray/test/common/*.hpp"
+   "include/detray/test/common/detail/*.hpp"
+   "include/detray/test/common/utils/*.hpp")
 
-add_library( detray_test INTERFACE "${_detray_test_headers}" )
-add_library( detray::test ALIAS detray_test )
+add_library( detray_test_common INTERFACE "${_detray_test_headers}" )
+add_library( detray::test_common ALIAS detray_test_common )
 
-target_include_directories( detray_test
+target_include_directories( detray_test_common
    INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/include" )
 
-target_link_libraries( detray_test INTERFACE vecmem::core detray::core detray::csv_io detray::svgtools )
+target_link_libraries( detray_test_common INTERFACE GTest::gtest GTest::gtest_main vecmem::core detray::core detray::csv_io detray::svgtools )
 
 unset( _detray_test_headers )
 
+# Set up the specific device test libraries
+if( DETRAY_BUILD_CUDA OR DETRAY_BUILD_SYCL )
```

Comment:
You have individual checks inside of that `CMakeLists.txt` file as well for CUDA and SYCL. You should remove the check from here. With this, if we at one point add HIP tests, we'll have to modify 2 files instead of just 1.

Reply:
Removed